### PR TITLE
fix: non-JSON response body display for upload errors

### PR DESCRIPTION
**Pull request by cacieprins**
- Closes 

### Additional details

Upload endpoints can return response bodies that are not JSON when an HTTP error occurs. This PR ensures we properly parse response bodies, display them, and include them in the error report sent to the Cloud API.

### How has the user experience changed?
Before:
Screenshot 2024-07-03 at 11 46 20 AM
After:
Screenshot 2024-07-03 at 11 45 55 AM

### PR Tasks

- [X] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: system-tests/__snapshots__/record_spec.js [hunk 1]
```diff
@@ -3049,6 +3049,16 @@ exports['e2e record capture-protocol enabled protocol runtime errors db size too
...
+Warning: We encountered an HTTP error while uploading the Test Replay recording for this spec.
+
+These results will not display Test Replay recordings.
+
+This error will not affect or change the exit code.
+
+undefined responded with HTTP undefined: undefined
+
+undefined
+
```
Comment (cacieprins) [inline system-tests/__snapshots__/record_spec.js]:
```suggestion
```

File: cli/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@ _Released 7/16/2024 (PENDING)_
 **Bugfixes:**
 
 - Fixed an issue where the ReadStream used to upload a Test Replay recording could erroneously be re-used when retrying in cases of retryable upload failures. Fixes [#29227](https://github.com/cypress-io/cypress/issues/29227)
+- CLI output properly displays non-JSON response bodies when a Test Replay upload attempt returns a non-JSON response body for a non-200 status code. Addressed in [#29801](https://github.com/cypress-io/cypress/pull/29801).
```
Comment (AtofStryker) [inline cli/CHANGELOG.md:8]: should this be marked as a bugfix or more of an improvement?

Comment (cacieprins) [reply cli/CHANGELOG.md:8]: I think it's a bug that we weren't appropriately handling non-JSON response bodies before this.

Comment (AtofStryker) [reply cli/CHANGELOG.md:8]: That makes sense. My main concern is that the `verify-release-readiness` works correctly here which I believe only runs on the `develop` branch. Are you able to verify there are no changelog issues if you run the job locally?

File: packages/server/lib/cloud/artifacts/upload_artifacts.ts [hunk 3]
```diff
@@ -16,6 +16,14 @@ import { NetworkError } from '../network/network_error'
 
 const debug = Debug('cypress:server:cloud:artifacts')
 
+const rmWspAndTrim = (str: string) => {
```
Comment (mschile) [inline packages/server/lib/cloud/artifacts/upload_artifacts.ts]: Does `rmWsp` stand for `removeWhitespace`? If so, suggest just spelling it out:

```suggestion
const removeWhitespaceAndTrim = (str: string) => {
```

**Review by mschile — APPROVED** [review]
Looks good overall! Just added one small suggestion.

**Review by AtofStryker — APPROVED** [review]

**Pull request merged**